src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -798,12 +709,20 @@ public int getCurrentPoolSize() {
      * @return current number of workers executing tasks
      */
     public int getCurrentRunningCount() {
-      synchronized (workersLock) {
-        int workerOutQty = currentPoolSize - availableWorkers.size();
-        if (waitingForWorker || currentPoolSize == 0) {
-          return workerOutQty;
+      int startPoolSize = currentPoolSize.get();
+      while (true) {
+        int idleWorkerCount = 0;
+        Worker idleWorker = this.idleWorker.get();
+        while (idleWorker != null) {
+          idleWorkerCount++;
+          idleWorker = idleWorker.nextIdleWorker;
+        }
+      
+        if (currentPoolSize.get() == startPoolSize) {
+          // TODO - do we want to re-check if the idle queue changed during computation
```

Comment:
This is a TODO note I have not decided on.  The question is how much do we care that we have read a stable state.  Right now we just ensure the pool has not grown, but additional threads may have become idle while we were totaling the idle threads.